Feature/cutscenes

Toine2106 opened the pull request.

Toine2106 committed:
New Cutscene sprites and animation added new scene for intro cutscene

Toine2106 committed:
Added outo cutscene to the game

Studeeric committed:
Simplified csm with an enum

Toine2106 committed:
church cutscene

Toine2106 committed:
Merge branch 'feature/cutscenes' of https://github.com/smooth-soup-studios/oh-my-cult into feature/cutscenes

Toine2106 committed:
final changes and fixes

hexaclue commented on Assets/Scripts/Managers/CutsceneManager.cs:
```diff
@@ -0,0 +1,27 @@
+
+public class CutsceneManager : MonoBehaviour {
+	// Start is called before the first frame update
+	private enum CutScene {
+		Intro = 14, // Name is humanname and number is duration in seconds.
+		Church = 7,
+		Outro = 12
+	}
```
That's NOT how you should use enums 😓

Studeeric replied:
Hoe dan wel? This is not how you submit feedback. When submitting feedback you clarify why something is wrong, this is not what has happened here. It just states what is wrong but not why. :D

hexaclue replied:
Enums are actually used to specify a unique value. Might someone need another cutscene with the same duration as one of these, that wouldn't be possible. Better would be to use a Dictionary, create a new data class with a corresponding list or just use a simple list with durations in the case of fixed cutscene IDs. Another way would be to use list of tuples, containing the name (`string`) in value 1, and the duration (`float`) in value 2.

Studeeric replied:
Fair, had alleen geen zin om t uit te zoeken hoe ookalweer een dictionary werkte voor dit simpele eenmalige probleem. hihi

Justwouter replied:
Agreed hahaha

hexaclue commented on Assets/Scripts/Managers/Doors/DoorController.cs:
```diff
@@ -62,7 +62,7 @@ public class DoorController : MonoBehaviour {
 					TransportTo switch {
 						TransportDestination.House => SceneDefs.HouseInteriorLevel,
-						TransportDestination.Special => SceneDefs.BossArenaLevel,
+						TransportDestination.Special => SceneDefs.ChurchCutscene,
```
Works for now but I don't like this. Could've added a new TransportDestination (e.g. "Cutscene")

hexaclue commented on Assets/Scripts/Controllers/AI/Boss/Boss.cs:
```diff
@@ -37,7 +36,7 @@ public class Boss : MonoBehaviour, ISaveable {
-				SceneManager.LoadScene(SceneDefs.EndingScreen);
+				SceneManager.LoadScene(SceneDefs.OutroCutscene);
 			}
 		});
```
1. Consider putting this into a new method that isn't anonymous/lambda.
2. Could use some sort of fade-out animation (for the expo), but this works for now.

Toine2106 replied:
Why does it need a new methode i just changed the scene name. when we didn't had the cutscene it was fine but now it needs a new method. Why?

hexaclue replied:
Just a general code review. To me, it would be just a bit more nice to read (less cluttered).

hexaclue reviewed (dismissed):
Requested change:
- Boss.cs, comment 1
- CutsceneManager.cs

Possible change:
- Boss.cs, comment 2
- DoorController.cs

Justwouter committed:
Small fixes
- Fixed nullref on HUD disable
- Removed weird enum stuff in CutsceneManager
- Added seperate death method to boss instead of lambda
-Removed tutorial scene

Justwouter committed:
Merge branch 'dev' into feature/cutscenes

Justwouter committed:
Merge branch 'dev' into feature/cutscenes
